docs/sdk-json-proposal/SDK_API_DEFINITION.md

```diff
@@ -0,0 +1,257 @@
+# Mesh Data Plane SDK Definition
+
+## mesh_create_client()
+```c
+int mesh_create_client(MeshClient **client,
+                       const char *config_json)
+```
+Creates a new mesh client with the provided JSON configuration.
+
+#### Parameters
+* `[OUT]` `client` – Address of a pointer to a mesh client structure.
+* `[IN]` `config_json` – Pointer to a mesh client configuration string in JSON format.
+
+#### Client Configuration Example
+```json
+{
+  "apiVersion": "v1",
+  "apiConnectionString": "Server=192.168.95.1; Port=8001",
+  "apiDefaultTimeoutMicroseconds": 100000,
+  "maxMediaConnections": 32,
+}
+```
+
+#### Returns
+0 if successful. Otherwise, returns an error.
+
+
+## mesh_delete_client()
+```c
+int mesh_delete_client(MeshClient **client)
+```
+Deletes the mesh client and its resources.
+
+#### Parameters
+* `[IN/OUT]` `client` – Address of a pointer to a mesh client structure.
+
+#### Returns
+0 if successful. Otherwise, returns an error.
+
+
+## mesh_create_tx_connection()
+```c
+int mesh_create_tx_connection(MeshClient *client,
+                              MeshConnection **conn,
+                              const char *config_json)
+```
+Creates a transmitter connection with the provided JSON configuration.
+
+#### Parameters
+* `[IN]` `client` – Pointer to a parent mesh client.
+* `[OUT]` `conn` – Address of a pointer to the connection structure.
+* `[IN]` `config_json` – Pointer to a connection configuration structure.
+
+#### Transmitter/Receiver Connection Configuration Example
+```json
+{
+  "bufferQueueCapacity": 16,
+  "maxPayloadSize": 2097152,
+  "maxMetadataSize": 8192,
+  "connType": "multipoint-group" | "mediabuf" | "st2110" | "rdma",
+  "connection": {
+    "multipoint-group": {
+      "ipAddr": "224.0.0.1",
+      "port": "9003",
+      "urn": null, // can be used instead of ipAddr+port
+    },
+    "mediabuf": {
```

Comment:
what about to change it to something more specific e.g. local-memif?

Reply:
what when we switch from memif to another local communication mechanism?

Reply:
This mediabuf section is to be removed because of removing horizontal communication between apps skipping mediaproxy.

Reply:
Please remove this section from JSON file.